### DISCUSSION TITLE
ModPlug_Tell symbol is missing

### DISCUSSION THREAD

**Discussion by JeodC** [dxx-rebirth/dxx-rebirth#767] [Q&A]
See https://github.com/PortsMaster/PortMaster-New/issues/40

#### Description
The ports can't use the .hmp files and soundfonts. Expected behavior is SDL2_Mixer will decode .hmp and send to libmodplug for playback. However, libmodplug fails to find the symbol `MODPLUG_TELL` and as a result there is no music unless a .dxa addon file is used. This is potentially a pipewire issue as bringing over libmodplug from known working cfws does not help. Other firmwares use pulse audio and do not call libmodplug.

Attempted:
- lib32 pipewire exports
- `export SDLAUDIODRIVER=pipewire`

Debug.txt relevant section:
```
`ModPlug_SetMasterVolume'
     64354:	symbol=ModPlug_GetName;  lookup in file=/roms/ports/descent/libs/libmodplug.so.1.0.0 [0]
     64354:	binding file /roms/ports/descent/libs/libmodplug.so.1.0.0 [0] to /roms/ports/descent/libs/libmodplug.so.1.0.0 [0]: normal symbol `ModPlug_GetName'
     64354:	symbol=ModPlug_Tell;  lookup in file=/roms/ports/descent/libs/libmodplug.so.1.0.0 [0]
     64354:	symbol=ModPlug_Tell;  lookup in file=/usr/lib/libstdc++.so.6 [0]
     64354:	symbol=ModPlug_Tell;  lookup in file=/usr/lib/libm.so.6 [0]
     64354:	symbol=ModPlug_Tell;  lookup in file=/usr/lib/libc.so.6 [0]
     64354:	symbol=ModPlug_Tell;  lookup in file=/lib/ld-linux-aarch64.so.1 [0]
     64354:	symbol=ModPlug_Tell;  lookup in file=/usr/lib/libgcc_s.so.1 [0]
     64354:	/roms/ports/descent/libs/libmodplug.so.1.0.0: error: symbol lookup error: undefined symbol: ModPlug_Tell (fatal)
```

As far as I can tell, neither [sdl_mixer](https://github.com/SDL-mirror/SDL_mixer/blob/2d53d383a63293aff1249d525ff978e3a4dac002/src/codecs/music_modplug.c) or [sdl2_mixer](https://github.com/emscripten-ports/SDL2_mixer/blob/master/music_modplug.c) define `ModPlug_Tell`. I am at a loss for whether this is a Rebirth issue or a library issue--which is why this is not a formal issue, Any insights welcome.

**Comment by JeodC** [comment]
I believe I have answered my own question: https://github.com/Konstanty/libmodplug/pull/47

**Comment by vLKp** [comment]
Rebirth has no matches for `ModPlug` anywhere in the source tree.  Rebirth does not compose strings for symbol lookup, so I think the undefined reference does not come from Rebirth.  The linked pull request seems to be about adding this feature to libmodplug, so I think that cannot explain the undefined reference.  I see that SDL2_Mixer can be built with support for libmodplug.  Your output seems to indicate that libmodplug is the origin of the undefined reference, which makes no sense to me.  If libmodplug lacks that feature (because https://github.com/Konstanty/libmodplug/pull/47 was never accepted), then it should not reference the symbols from that feature.  If your libmodplug has that feature patched in, then the symbols should exist.

From where did you get your SDL2_Mixer and libmodplug?  What is the output of `nm -D --defined-only /roms/ports/descent/libs/libmodplug.so.1.0.0`?

**Reply by JeodC** [reply]
SDL2_Mixer is prebuilt and packaged with the firmware. In JelOS' case, it is grabbed as a package: https://github.com/JustEnoughLinuxOS/distribution/blob/dev/packages/devel/SDL2_mixer/package.mk

**Comment by JeodC** [comment]
I am currently waiting for others to test the libmodplug fork I built to see if it resolves the issue.
